tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/parallel/PathUpdateTests.java

```diff
@@ -0,0 +1,333 @@
+                Json.createObjectBuilder().add("cm:name", newName)).build();
+        return restClient.authenticateUser(user).withCoreAPI().usingNode(node).updateNode(renameJson.toString());
+    }
+
+    /**
+     * Move the specified node to a folder.
+     *
+     * @param restClient The REST client to use to send the request.
+     * @param user The user to use for the move.
+     * @param node The node to move.
+     * @param targetFolder The folder to move the node to.
+     * @return The updated node.
+     */
+    private RestNodeModel moveNode(RestWrapper restClient, UserModel user, ContentModel node, FolderModel targetFolder)
+    {
+        RestNodeBodyMoveCopyModel moveBody = new RestNodeBodyMoveCopyModel();
+        moveBody.setTargetParentId(targetFolder.getNodeRef());
+        return restClient.authenticateUser(user).withCoreAPI().usingNode(node).move(moveBody);
+    }
+
+    /**
+     * Create a category.
+     *
+     * @param restClient The REST client to use.
+     * @param user The user to use to create the category.
+     * @param parentId The parent in which to create the category.
+     * @return The new category.
+     */
+    private RestCategoryModel createCategory(RestWrapper restClient, UserModel user, String parentId)
+    {
+        return restClient.authenticateUser(user)
+                         .withCoreAPI()
+                         .usingCategory(RestCategoryModel.builder().id(parentId).create())
+                         .createSingleCategory(RestCategoryModel.builder().name(RandomData.getRandomAlphanumeric()).create());
+    }
+
+    /**
+     * Create a path to a file or folder in a site.
+     *
+     * @param site The site object.
+     * @param documentLibraryNames The list of names of nodes from the document library to the target file or folder.
+     * @return An absolute path suitable for use in a path query.
+     */
+    private String pathInSite(SiteModel site, String... documentLibraryNames)
+    {
+        return "/app:company_home/st:sites/cm:"+ site.getId() + "/cm:documentLibrary/cm:" + stream(documentLibraryNames).collect(joining("/cm:"));
+    }
+
+    /**
+     * Create a path to a file or folder via the category hierarchy.
+     *
+     * @param nodeNames The ordered list of node names from the root category to the node that was categorised.
+     * @return An absolute path through the categories to the specified node.
+     */
+    private String categoryPath(String... nodeNames)
+    {
+        return "/cm:categoryRoot/cm:generalclassifiable/cm:" + Arrays.stream(nodeNames).collect(joining("/cm:"));
+    }
+
+    /**
+     * Wait for a second. This is needed due to the index refresh interval - see ACS-4637.
+     */
+    private void waitForOneSecond()
+    {
+        try
+        {
+            TimeUnit.SECONDS.sleep(1);
+        }
+        catch (InterruptedException e)
+        {
+            LOGGER.warn("Failed to wait for full second.");
```

Comment:
I'm not sure if this is a general rule but at least in multi-threaded apps (don't know how the parallelism works here) there was a rule of thumb not to ignore `InterruptedException` and IIRC, as the simplest mean just call `Thread.currentThread().interrupt();`.

Reply:
I was able to remove these sleep requests completely as we have fixed the underlying issue in the ESC.